tests/test_slim_engine.py

```diff
@@ -3079,6 +3079,79 @@ def test_errors(self):
             stdpopsim.selection_coeff_from_mutation(ts, "bar")
 
 
+@pytest.mark.skipif(IS_WINDOWS, reason="SLiM not available on windows")
+class TestPopSizes:
+    """
+    Test that population sizes are set correctly.
+    Thanks to discretization artifacts, this is a bit tricky in phases
+    of exponential growth: suppose we are growing at rate r for the
+    time period [t, t+T), starting from size x. In discrete time, at
+    time t we set the size to x; then subsequently we re-set the size
+    only T-1 times. (One consideration is that we re-set sizes in `late`,
+    which doesn't take effect until the next tick, so a reset in the final
+    time step has no effect.) This means that the continuous-time msprime model
+    that we base this on is growing for total time T, but the SLiM model grows
+    in T-1 time steps. Furthermore, we'd like both the final and initial sizes
+    to be exactly what they are supposed to be. This amounts to approximating
+    exp(r*t) over the interval [0,T] by exp(r*t*T/(T-1)). Probably something
+    nicer could be done, but this is close. The approximation is better for
+    smaller growth rates.
```

Comment:
It won't tell you anything you don't already know, but note that the SLiM manual, section 5.1.2, has a extended (probably too extended!) discussion of exactly these sorts of issues for exponential growth.  Possibly there would be something useful for you there.

Reply:
or possibly you will tell me that I should rewrite that section in a less dumb/boring way, which is also fine :->

Reply:
I just learned some things from that! 
![Screenshot from 2024-12-18 21-16-05](https://github.com/user-attachments/assets/e9ac769b-1eb4-4d90-a354-a1c078925f4d)